AdyenComponents/Apple Pay/ApplePayPayment.swift

```diff
@@ -0,0 +1,98 @@
+//
+// Copyright (c) 2022 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Adyen
+import Foundation
+import PassKit
+
+// MARK: - Apple Pay component configuration.
+
+/// Describes the Apple Pay payment.
+public struct ApplePayPayment {
+
+    /// Create a new instance of ApplePayPayment.
+    /// - Parameters:
+    ///   - countryCode: The code of the country in which the payment is made.
+    ///   - currencyCode: The code of the currency in which the amount's value is specified.
+    ///   - summaryItems: The summary items in a payment request—for example, total, tax, discount, or grand total.
+    /// - Throws: `ApplePayComponent.Error.emptySummaryItems` if the summaryItems array is empty.
+    /// - Throws: `ApplePayComponent.Error.negativeGrandTotal` if the grand total is negative.
+    /// - Throws: `ApplePayComponent.Error.invalidSummaryItem` if at least one of the summary items has an invalid amount.
+    /// - Throws: `ApplePayComponent.Error.invalidCountryCode` if the `payment.countryCode` is not a valid ISO country code.
+    /// - Throws: `ApplePayComponent.Error.invalidCurrencyCode` if the `Amount.currencyCode` is not a valid ISO currency code.
+    public init(countryCode: String,
+                currencyCode: String,
+                summaryItems: [PKPaymentSummaryItem]) throws {
+        guard CountryCodeValidator().isValid(countryCode) else {
```

Comment:
i wonder if we could make a throwable `isValid` function for `Validator`. and make validators for every condition we have here. we could end up with something like 
```swift
try CurrencyCodeValidator().isValid(currencyCode, onFail: ApplePayComponent.Error.invalidCountryCode)
try CountryCodeValidator().isValid(countryCode, onFail: ApplePayComponent.Error.invalidCurrencyCode)
try NonEmptyValidator(summaryItems, onFail: ApplePayComponent.Error.emptySummaryItems)
```
but that's probably too much :)

Reply:
Maybe another PR 😅